### ISSUE TITLE
Timezone handling

### ISSUE THREAD

**Issue by thekid** [xp-forge/handlebars-templates#3]
Currently, dates are displayed as-is in the frontend. However, the user viewing the frontend may be in a different timezone, and in case we're using timestamps, most commonly will be.

One way would be to pass the timezone as optional argument:

```handlebars
{{date created timezone="Europe/Berlin"}}
{{date created timezone=user.tz}}
```

However, a default timezone may be a good idea. It could cascade as follows:

* If we have a logged-in user, we could determine the timezone from there
* In case of unauthenticated requests, we could use GeoIP, falling back to a default timezone for the frontend

**Comment by thekid** [comment]
See also:

* https://twig.symfony.com/doc/3.x/filters/date.html#timezone states: *By default, the date is displayed by applying the default timezone (the one specified in php.ini or declared in Twig – see below), but you can override it by explicitly specifying a timezone*
* https://www.npmjs.com/package/handlebars-i18n uses `timeZone=...` & [Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat)
* https://vuejsexamples.com/datetime-formatting-parsing-in-vue-using-luxon/ shows how VueJS and Luxon handle time zones -both as optional parameter on placeholders *and* as default

**Comment by thekid** [comment]
Idea:

```php
$local= TimeZone::getLocal();

$templates= new Handlebars($path);
$templates->output(
  timezone: fn($req) => $req->value('user')['tz'] ?? $local,
  dateformats: ['day' => 'd.m.Y']
);
```

**Comment by thekid** [comment]
Modern browsers support this:

```javascript
Intl.DateTimeFormat().resolvedOptions().timeZone;
// "Europe/Berlin" in my case
```

*Source: https://stackoverflow.com/questions/13/determine-a-users-timezone*

This could be transmitted to the server in a cookie, by redirect or by XHR but would not work for the *first* request to the page, thus requiring a redirect.

**Comment by thekid** [comment]
Idea based on `use` method which accepts a collection of helpers:

```php
$local= TimeZone::getLocal();

$templates= new Handlebars($path);
$templates->use(new Dates(
  timezone : fn($req) => $req->value('user')['tz'] ?? $local,
  formats  : ['date' => 'd.m.Y', 'time' => 'H:i:s', 'datetime' => 'd.m.Y H:i'],
));
```

**Comment by thekid** [comment]
Basic functionality of using a server-side default timezone implemented by #4 , xp-forge/handlebars-templates@de77af5 and xp-forge/handlebars-templates@ce196a1 - extracting the timezone from the request will be handled separately.
